File: fsharp/bob/Solution/Bob/Bob.fs
module Bob
open System
open System.Text.RegularExpressions

let (|RegexMatch|_|) pattern input =
    match Regex.IsMatch(input, pattern) with
    | true -> Some true
    | false -> None
    
let hey message =
    match message with
    | RegexMatch "!$|^(?:[A-Z]+\s*)+[!.?]?$" message -> "Whoa, chill out!"
    | RegexMatch "\?$" message -> "Sure."
    | RegexMatch "^\s*$" message ->"Fine. Be that way!"
    | _ -> "Whatever." 

File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(message){
    var response = "Whatever.";

    if (message == message.toUpperCase() && message.match(/[A-Z]/))
    {
        response = "Whoa, chill out!";
    }
    else if (message.substr(-1)=="?")
    {
        response = "Sure.";
    }
    else if(message.trim() == "") 
    {
        response = "Fine. Be that way!";
    }

    return response;
};




module.exports = Bob;